### PR TITLE
Split docs build and link check

### PR DESCRIPTION
**Pull request by kazaamjt**
# Description

adds a seperate test that builds the docs and checks the returncode.
Using a mark only the build test can be called.
Replaced the original tox doc build test to call this.
Add an env to tox that performs both the build and linkcheck.

Should I add an entry to teh changelog?

Needed for https://github.com/inmanta/infra-tickets/issues/111
after this passed review, a nightly build needs to be configured to run the link-check

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] ~~Attached issue to pull request~~
- [ ] ~~Changelog entry~~
- [ ] ~~Type annotations are present~~
- [x] Code is clear and sufficiently documented
- [ ] ~~No (preventable) type errors (check using make mypy or make mypy-diff)~~
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: docs/check_sphinx.py [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
     docs_dir = os.path.dirname(os.path.abspath(__file__))
     # Ensure that all required files such as inmanta.pdf exist
-    subprocess.check_call(["make", "-C", docs_dir, "html", f"BUILDDIR={tmpdir}"])
-    return docs_dir, doctrees, htmldir
+    build_proc = subprocess.run(["make", "-C", docs_dir, "html", f"BUILDDIR={tmpdir}"], check=False)
+    return docs_dir, doctrees, htmldir, build_proc
+
+
+@pytest.mark.build
+@pytest.mark.dependency()
```
Comment (bartv) [inline docs/check_sphinx.py:19]: why is this required?

Comment (kazaamjt) [reply docs/check_sphinx.py:19]: For pytest_dependency. Marking it like so makes the result of the test internally available so other tests can depend on it.

Comment (bartv) [reply docs/check_sphinx.py:19]: I am not sure it is that important that on failure of the build the linkcheck is not executed. This means we would get fail,fail instead of fail,skip. That is not enough reason for me to add an additional dep to the requirements.

Comment (kazaamjt) [reply docs/check_sphinx.py:19]: Ok I will take it out. It doesn't make a big difference. The link-check doesn't take that long anyway.

**Review by bartv — APPROVED** [review]
